home/zsh/prompt-vcs-info.zsh

```diff
@@ -13,76 +13,79 @@ zstyle ':vcs_info:*' unstagedstr '🚧'
-# $1 will contain the 0-based index of the current format, so we can use that to ignore calls for formats we don't care about.
-#
+# Declare the async job and callback
+function async_job_update_git_untracked_skipped() {
+    cd $1
 
-function +vi-update-git-untracked() {
-    if [[ "$1" != "1" ]]; then
-            return 0
-    fi
+    local result=""
 
     # ZSH doesn't support showing untracked files. Adapted approach from here:
     # http://web.archive.org/web/20121018120253/https://briancarper.net/blog/570/git-info-in-your-zsh-prompt
     if [[ -n $(git ls-files --other --exclude-standard 2> /dev/null) ]]; then
-        hook_com[unstaged]+="👽"
-    fi
-}
-
-function +vi-update-git-skipped() {
-    if [[ "$1" != "1" ]]; then
-            return 0
+        result="${result}👽"
     fi
 
     # I wanted the following test to be `[[ -n $(git ls-files -v 2> /dev/null | grep ^S 2> /dev/null) ]]` but that always returns `false` when run from a vcs_info hook. I think it has something to do with the pipe (|). Relying on the git alias is working though.
     if [[ -n $(git li) ]]; then
-        hook_com[unstaged]+="🙈"
+        result="${result}🙈"
     fi
+
+    print $result
 }
 
-function _update_vcs_info() {
-    cd $1
-    vcs_info
-    
-    # vcs_info_msg_0_ will contain the branch info if we're in a git working tree and will be empty otherwise. No need to conditionally add this to the result
-    local result="${vcs_info_msg_0_}"
+function async_job_update_git_untracked_skipped_done() {
+    # https://github.com/mafredri/zsh-async?tab=readme-ov-file#async_process_results-worker_name-callback_function
+    local async_job_stdout=$3
+    local has_next_result_in_buffer=$6
+
+    update_git_prompt_string $async_job_stdout
 
-    # Both vcs_info_msg_1_ and vcs_info_msg_2_ will contain staged, unstaged, untracked and skipped status. These will always be set to at least "()", even if there is no status. Because of this we are only adding them to the result if their string length is greater than 2. I have some ideas on how to make this better, but this is working for now.
-    if [[ ${#vcs_info_msg_1_} -gt 2 ]]; then
-        result="${result} ${vcs_info_msg_1_}"
+    # Only reset the prompt if the async results buffer is empty
+    if [[ -n $has_next_result_in_buffer ]]; then
+        zle reset-prompt
     fi
+}
 
-    if [[ ${#vcs_info_msg_2_} -gt 2 ]]; then
-        result="${result} ${vcs_info_msg_2_}"
+async_init
+async_start_worker git_info_worker
+async_register_callback git_info_worker async_job_update_git_untracked_skipped_done
+
+function update_git_prompt_string() {
+    # When called from the async job callback, $1 will have the untracked and skipped status string. Otherwise it will be null or empty
+    local git_untracked_or_skipped_msg=$1
+
+    git_prompt_string=""
+    
+    # Branch
+    if [[ -n ${vcs_info_msg_0_} ]]; then
+        git_prompt_string="(${vcs_info_msg_0_})"
```

Comment:
Might be nice to declare some locals at the top for the vcs info messages here. Something like:

```
local git_branch_message=${vcs_info_msg_0_}
```

I'll probably do this next time I'm mucking around in here.